### ISSUE TITLE
Wrong permissions on local-backupbuckets

### ISSUE THREAD

**Issue by ShiroDN** [gardener/gardener#6752]
**How to categorize this issue?**

/area usability
/kind bug

**What happened**:
I am trying to setup local poc according to `getting_started_locally.md` guide. But it seems there is some problem with deployment on linux (archlinux, ubuntu 20.04). After I run `make gardener-up` the seed cluster never reaches `STATUS=Ready`.

```
tomas@gardener:~/gardener-1.56.1$ kubectl get seed local
NAME    STATUS     PROVIDER   REGION   AGE     VERSION      K8S VERSION
local   NotReady   local      local    7m18s   v0.0.0-dev   v1.24.0
```

When I run `kubectl describe seed local` I can see error (gardener-extension-provider-local-xxxx-yyyy):
```
Message:               The following BackupBuckets have issues:
* Name: c58c1901-0400-4725-a16a-d4ef9c200bcd, Error: flow "Backup Bucket Reconciliation" encountered task errors: [task "Waiting until backup bucket is reconciled" failed: Error while waiting for BackupBucket /c58c1901-0400-4725-a16a-d4ef9c200bcd to become ready: error during reconciliation: Error reconciling backupbucket: mkdir /dev/local-backupbuckets/c58c1901-0400-4725-a16a-d4ef9c200bcd: permission denied]
```

Then, if I try to create a shoot cluster with the command `kubectl apply -f example/provider-local/shoot.yaml` it is stuck at `Create Processing (26%)`:
```
tomas@gardener:~/gardener-1.56.1$ kubectl -n garden-local get shoot local
NAME    CLOUDPROFILE   PROVIDER   REGION   K8S VERSION   HIBERNATION   LAST OPERATION            STATUS      AGE
local   local          local      local    1.25.0        Awake         Create Processing (26%)   unhealthy   24m
```

After some debbugging of this problem I find out, that it's a permissions problem with hostpath `/etc/gardener/local-backupbuckets (dev/local-backupbuckets/)` which had the wrong owner id - id of my linux local user 1000:

```
root@gardener-local-control-plane:/etc/gardener/local-backupbuckets# ls -lah 
total 4.0K
drwxrwxr-x 2 1000 1000 4.0K Sep 29 07:44 .
drwxr-xr-x 5 root root   75 Sep 29 07:44 ..
```

Workaround is to `chmod o+w dev/local-backupbuckets/`

After that a bucket is created under user id 65532 and everything works just fine.
```
[tomas@w00:gardener-1.56.1]$ ls -lah dev/local-backupbuckets/
total 12K
drwxrwxrwx 3 1000 1000 4.0K Sep 29 07:46 ./
drwxr-xr-x 4 1000 1000 4.0K Sep 29 07:46 ../
drwxrwxr-x 2 65532 65532 4.0K Sep 29 07:46 1de2ce3a-fed8-420b-a826-ce49b4748ba4/
```

**What you expected to happen**:

It should just work.

**How to reproduce it (as minimally and precisely as possible)**:

On linux machine (tested on archlinux and ubuntu 20.04):
```
$ wget https://github.com/gardener/gardener/archive/refs/tags/v1.56.1.tar.gz
$ tar -xvf v1.56.1.tar.gz
$ cd gardener-1.56.1
$ make kind-up
$ make gardener-up
$ kubectl get seed local   --- will never reach STATUS=Ready
```
**Anything else we need to know?**:

**Environment**:

- Gardener version: 1.56.1
- Kubernetes version (use `kubectl version`): v1.24.3
- Cloud provider or hardware configuration: local
- Others:

**Comment by rfranzke** [comment]
Do you want to file a PR with an improvement? :)

**Comment by ShiroDN** [comment]
So, I did some tests and there is one more issue that is connected to this one.

I will file a PR but first I want to discuss how it should be best resolved.

The issue with permissions on dev/local-backupbuckets/ can be easily solved by a change in Makefile target `kind-up`:
`mkdir -m 775 ... -> mkdir -m 777 ...`

But the next issue is with deleting and cleanup.
In directories `local-backupbuckets` and `local-registry` under `dev/` there are files that are owned by root (id 0), these files can't be deleted by `kind-down`: `rm -rf dev/local-backupbuckets` or user without root privileges. The same thing applies for `local-registry`.

```
rm -rf dev/local-backupbuckets
rm: cannot remove 'dev/local-backupbuckets/6cb4d3df-1b94-4ce7-b460-20ba89b95b21/shoot--local--local--9580f33c-d60b-4c92-a78a-e8a139c4ee56': Permission denied
```

On mac this is not an issue, because docker on mac is using Linux VM, and files are shared from mac host to VM through gRPC fuse with user mapping.

I think this should be taken care of in some form, there are 3 different solutions that I came up with:

1. Best thing would be to create another make target like `kind-vol-cleanup` into which we would move deleting of `local-backupbuckets` too:

```
kind-vol-cleanup:
docker run --rm -v $(REPO_ROOT)/dev:/local-dev busybox sh -c 'rm -rf /local-dev/local-backupbuckets'
docker run --rm -v $(REPO_ROOT)/dev:/local-dev busybox sh -c 'rm -rf /local-dev/local-registry'
```
With this, we can work around the need of explicitly using sudo like it is done in `kind-up` for `fs.inotify.max_user_instances` increase.

2. We can use OS detection and if `Linux` is detected, then we can run rm commnad with sudo.
3. We can add a warning with a link to docs if `rm` returns a non-zero code.
